src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -170,11 +201,32 @@ public int allowedSpaceCount()
         }
     }
 
-    private List<String> getCitizenTabCompletions(final String potentialArgumentValue)
+    private List<String> getCitizenTabCompletions(@NotNull final TreeNode<IMenu> actionMenuTreeNode, final String potentialArgumentValue)
     {
         // TODO: see if we can figure out what citizen we are looking at as the default tab completion.
-        final List<String> citizenNameStrings = getCitizenNames();
-        final List<String> citizenNumberStrings = getColonyIdStrings();
+        final IMenu menu = actionMenuTreeNode.getData();
+        @Nullable Colony colony = null;
+        if (!menu.getMenuType().isNavigationMenu())
+        {
+            final ActionMenu actionMenu = (ActionMenu)menu;
+            // Try to find a valid colony value.
+            // TODO: doesn't check subarguments but should only check arguments for parents of this argument.
+            // TODO: Also no guarantee that we've grabbed the right colony argument if the command has more than one.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)